core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java

```diff
@@ -8221,6 +8221,26 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedSparkQuery));
   }
 
+  @Test public void testQualifyWithReferenceFromSelect() {
```

Comment:
this test is passing on master as well.

Reply:
this test case is added to show QUALIFY filter functionality did not change. added another test case which will fail in master because of previous handling of QUALIFY.

Reply:
okay